common/Util.cpp

```diff
@@ -113,6 +113,7 @@ namespace Util
         // N.B. Always reseed after getting forked!
         void reseed()
         {
+            std::unique_lock<std::mutex> lock(_rngMutex);
```

Comment:
Aha, if getNext() uses _rngMutex to access _rng, then we should do the same here.